src/Pagerfanta/Adapter/DoctrineORMAdapter.php

```diff
@@ -29,13 +30,16 @@ class DoctrineORMAdapter implements AdapterInterface
      * Constructor.
      *
      * @param \Doctrine\ORM\Query|\Doctrine\ORM\QueryBuilder $query               A Doctrine ORM query or query builder.
-     * @param boolean                                        $fetchJoinCollection Whether the query joins a collection (true by default).
-     * @param boolean|null                                   $useOutputWalkers    Whether to use output walkers pagination mode
+     * @param \Doctrine\ORM\EntityManager|null               $em                  Entity Manager
+     * @param Boolean                                        $fetchJoinCollection Whether the query joins a collection (true by default).
+     * @param Boolean|null                                   $useOutputWalkers    Whether to use output walkers pagination mode
      */
-    public function __construct($query, $fetchJoinCollection = true, $useOutputWalkers = null)
+    public function __construct($query, $em = null, $fetchJoinCollection = true, $useOutputWalkers = null)
```

Comment:
this is a BC break for anyone using that class. Adding it in a non-final position is a no-go.